hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/HazelcastSqlValidator.java

```diff
@@ -563,13 +565,36 @@ public CalciteContextException newValidationError(SqlNode node, Resources.ExInst
             Object[] arguments = ResourceUtil.args(e);
             String identifier = (arguments != null && arguments.length > 0) ? String.valueOf(arguments[0]) : null;
             Mapping mapping = identifier != null ? iMapResolver.resolve(identifier) : null;
-            String sql = mapping != null ? SqlCreateMapping.unparse(mapping) : null;
+            String sql = mapping != null && hasMapAccess(identifier) ? SqlCreateMapping.unparse(mapping) : null;
             String message = sql != null ? ValidatorResource.imapNotMapped(e.str(), identifier, sql) : e.str();
             throw QueryException.error(SqlErrorCode.OBJECT_NOT_FOUND, message, exception, sql);
         }
         return exception;
     }
 
+
+    /**
+     * Check read and create permission for the map.
+     * This method does not throw an exception, but rather provides the results of a permission check.
+     * Use in scenarios where it is needed to check permissions without interrupting the process.
+     *
+     * @param map name of the map.
+     * @return {@code true} access is allowed, {@code false} otherwise.
+     */
+    private boolean hasMapAccess(String map) {
+        if (!ssc.isSecurityEnabled()) {
+            return true;
+        }
+        var permission = new MapPermission(map, ActionConstants.ACTION_CREATE, ActionConstants.ACTION_READ);
```

Comment:
```suggestion
        var permission = new MapPermission(map,  ActionConstants.ACTION_READ);
```
that way it will be consistent with SqlMappingDdlTask.getRequiredPermission(). Also `iMapResolver` never creates and IMap if it does not exist, so we do not have to require `ACTION_CREATE`